src/ops/ops.go

```diff
@@ -229,7 +229,7 @@ func (o *ops) SetBootOrder(device string) error {
 	o.log.Info("Setting efibootmgr to boot from disk")
 
 	// efi-system is installed onto partition 2
-	_, err = o.ExecPrivilegeCommand(o.logWriter, "efibootmgr", "-d", device, "-p", "2", "-c", "-L", "RHCOS", "-l", "\\EFI\\BOOT\\BOOTX64.EFI")
+	_, err = o.ExecPrivilegeCommand(o.logWriter, "efibootmgr", "-d", device, "-p", "2", "-c", "-L", "Red Hat Enterprise Linux", "-l", "\\EFI\\redhat\\shimx64.efi")
```

Comment:
Does the label change make any difference? It's technically RHCOS and not RHEL so I'm not sure why it was added to the PR

Reply:
Yes it does, shim looks at the file /boot/EFI/redhat/BOOTX64.CSV , to find the label it expects and the path it expects

Reply:
Reference: https://github.com/rhboot/shim/blob/a1170bb00a116783cc6623b403e785d86b2f97d7/README.fallback#L33-L46 (I cannot understand from it if the label is required tbh).

Reply:
Thank you for the link.

> if it doesn't see the entry it expects it creates a new one

I went over the bugzilla / shim code / README.fallback and I'm not sure that's what's happening. The way I understood it, if the shim (which is present as both `BOOT/BOOTX64.efi` and `redhat/shimx64.efi`) gets booted as `BOOT/BOOTx64.efi` (rather than the correct `redhat/shimx64.efi`, and it sees that the fallback `BOOT/fbx64.efi` bootloader exists - it assumes that it happened because the uefi nvram boot entry got corrupt for some reason and as a result the firmware is trying to treat the disk in the default manner of booting `BOOT/BOOTx64.efi`.

In response to that, the shim runs the fallback bootloader `BOOT/fbx64.efi` whose purpose is to re-create the boot entry using the information present in `redhat/BOOTX64.CSV`.

Doesn't seem like it compares the label at any point to trigger this behavior, so it leads me to believe that the choice of label created by AI is of no consequence.

However, I guess if it falls-back to `Red Hat Enterprise Linux` on corruption, we might as well use the same name from the get go, even though it's a bit misleading since it's RHCOS and not RHEL

Reply:
@flaper87 , @celebdor please take a look